### DISCUSSION TITLE
Remote `.tflint.hcl` configs

### DISCUSSION THREAD

**Discussion by evairmarinho** [terraform-linters/tflint#1720] [Ideas]
I would love to have a common centralized `.tflint.hcl` for all my Terraform projects. TFlint could give the option to pull configuration from a remote git repository. However, I would like TFLint to be able to merge the "global" settings with specific settings for each Terraform project, in case it is necessary to enable or disable a rule.

```hcl
config {
  source = "https://github.com/my-org/devops-global-configs/blob/main/.tflint.hcl"
}

# Additional Rule that override "global" configuration. Only a example.
rule "terraform_standard_module_structure" {
    enabled = false
}
```

**Comment by evairmarinho** [comment]
In tfsec there is an implementation of this via CLI, via the parameter `--config-file-url` . 

Ref of implementation: https://github.com/aquasecurity/tfsec/blob/master/internal/app/tfsec/cmd/flags.go . Search for `configFileUrl`

**Comment by wata727** [comment]
I don't dislike the idea, but I'm concerned about compatibility with lockfile. See https://github.com/terraform-linters/tflint/issues/1634
If the config file is automatically updated, it may become inconsistent with the lockfile and cause `tflint --init` to fail.

For now, if you're using GitHub Actions, you can use tflint-load-config-action.
https://github.com/terraform-linters/tflint-load-config-action
